xchange-gateio/src/main/java/org/knowm/xchange/gateio/service/GateioMarketDataServiceRaw.java

```diff
@@ -106,4 +105,35 @@ public GateioTradeHistory getBTERTradeHistorySince(
     List<CurrencyPair> currencyPairs = new ArrayList<>(bter.getPairs().getPairs());
     return currencyPairs;
   }
+
+  public List<GateioKline> getKlines(CurrencyPair pair, GateioKlineInterval interval, int limit) throws IOException {
+
+    limit = limit < 0 ? 1 : limit;
+
+    long limitHours = (interval.getSeconds() * limit) / HOURS.toSeconds(1);
```

Comment:
if what you need is always hours just pass in limit hours instead of doing all this?

Reply:
Ok, 'limit' parameter replaced by 'hours' parameter.
Also added method without 'hours' parameter.

Reply:
what is the point of that second method?

Reply:
Just for info that it's not necessary to put parameter 'hours'.

Reply:
can we just eliminate it? I can approve once you do, not really worth having the function will exception always?